src/ring_theory/graded_algebra/basic.lean

```diff
@@ -103,6 +103,37 @@ dfinsupp.mem_support_iff.trans add_submonoid_class.coe_eq_zero.not.symm
 
 end graded_ring
 
+section add_left_cancel_monoid
```

Comment:
How painful is it to prove the `right` version too? Does using `mul_opposite` help at all?

Reply:
Assuming you mean
```lean
lemma direct_sum.decompose_mul_add_of_right_mem {ι σ A}
  [decidable_eq ι] [add_right_cancel_monoid ι] [semiring A]
  [set_like σ A] [add_submonoid_class σ A] (𝒜 : ι → σ) [graded_ring 𝒜]
  {i j : ι} {a : A} (b : 𝒜 j) :
  decompose 𝒜 (a * ↑b) (i + j) =
    @graded_monoid.ghas_mul.mul ι (λ i, 𝒜 i) _ _ _ _ (decompose 𝒜 a i) b
```
not hard at all, I have pushed an update